src/test/java/ch/wisv/areafiftylan/IntegrationTest.java

```diff
@@ -40,30 +41,60 @@
     protected UserRepository userRepository;
 
     protected User user;
+    protected final String userCleartextPassword = "password";
 
     protected User admin;
+    protected final String adminCleartextPassword = "password";
+
+    protected User outsider;
+    protected final String outsiderCleartextPassword = "password";
+
 
     SessionFilter sessionFilter = new SessionFilter();
 
     @Before
     public void initIntegrationTest() {
-        user = new User("user", new BCryptPasswordEncoder().encode("password"), "user@mail.com");
+        userRepository.deleteAll();
+
+        user = makeUser();
+        admin = makeAdmin();
+        outsider = makeOutsider();
+
+        userRepository.saveAndFlush(user);
+        userRepository.saveAndFlush(admin);
+        userRepository.saveAndFlush(outsider);
+
+        RestAssured.port = port;
```

Comment:
The test instance is started on a random port, so you can run and test at the same time. This binds the dynamic port to the test framework so that it works.